src/boot/efi/boot.c

```diff
@@ -436,8 +439,16 @@ static void ps_bool(const CHAR16 *fmt, BOOLEAN value) {
         Print(fmt, yes_no(value));
 }
 
-static void print_status(Config *config, CHAR16 *loaded_image_path) {
+static BOOLEAN ps_pause(void) {
         UINT64 key;
+        EFI_STATUS err;
+
+        Print(L"\n--- Press any key to continue, ESC or q to quit. ---\n\n");
+        err = console_key_read(&key, UINT64_MAX);
+        return (EFI_ERROR(err) || IN_SET(key, KEYPRESS(0, SCAN_ESC, 0), KEYPRESS(0, 0, 'q'), KEYPRESS(0, 0, 'Q')));
+}
```

Comment:
it looks a bit weird to return true on "quit" and false on "continue". Should be the other way round philopsophically?